### PR TITLE
Fix/AC-891 added check for files are coming from transcoder laravel API

### PR DESCRIPTION
**Pull request by DevikVekariya**
added check based on IP addresses.

Applied `caching` for IP.

Added `ignore_cache` check. it will help when we change our server.

### CODE REVIEW

File: admin/rt-transcoder-handler.php [hunk 1]
```diff
@@ -991,6 +991,31 @@ public function handle_callback() {
 		$thumbnail		= filter_input( INPUT_POST, 'thumbnail', FILTER_SANITIZE_STRING );
 		$format			= filter_input( INPUT_POST, 'format', FILTER_SANITIZE_STRING );
 
+		$ignore_cache = filter_input( INPUT_POST, 'ignore_cache', FILTER_SANITIZE_STRING );
+
+		if ( ! empty( $job_id ) ) {
+
+			// It returns false if cache is not set.
```
Comment (chandrapatel) [inline admin/rt-transcoder-handler.php]: `// It returns false if transient is not set.`

File: admin/rt-transcoder-handler.php [hunk 2]
```diff
@@ -991,6 +991,31 @@ public function handle_callback() {
...
+			$server_addr = get_transient( 'rtt_server_addr' );
+			if ( ( ! empty( $ignore_cache ) && $ignore_cache ) || false === $server_addr ) {
```
Comment (chandrapatel) [inline admin/rt-transcoder-handler.php]: Here, `! empty( $ignore_cache )` should enough. 

`if ( ! empty( $ignore_cache ) || false === $server_addr ) {`

File: admin/rt-transcoder-handler.php [hunk 3]
```diff
@@ -991,6 +991,31 @@ public function handle_callback() {
...
+
+				$path        = str_replace( 'http://', '', $this->transcoding_api_url );
+				$path_arr    = explode( '/', $path );
+				$server_addr = gethostbyname( $path_arr[0] );
+				set_transient( 'rtt_server_addr', $server_addr, 7 * 24 * 60 * 60 );
+
+			}
+
+			$incomming_addr = $_SERVER['REMOTE_ADDR']; // @codingStandardsIgnoreLine
```
Comment (chandrapatel) [inline admin/rt-transcoder-handler.php]: Use `filter_input()` and remove `// @codingStandardsIgnoreLine`

File: admin/rt-transcoder-handler.php [hunk 4]
```diff
@@ -991,6 +991,31 @@ public function handle_callback() {
...
+
+			if ( empty( $server_addr ) || empty( $incomming_addr ) || $incomming_addr !== $server_addr ) {
+
+				echo 'Something went wrong. Invalid post request.';
```
Comment (chandrapatel) [inline admin/rt-transcoder-handler.php]: Use `__()` function here.

File: admin/rt-transcoder-handler.php [hunk 5]
```diff
@@ -991,6 +991,31 @@ public function handle_callback() {
...
+		if ( ! empty( $job_id ) ) {
+
+			// It returns false if cache is not set.
+			$server_addr = get_transient( 'rtt_server_addr' );
+			if ( ( ! empty( $ignore_cache ) && $ignore_cache ) || false === $server_addr ) {
+
+				$path        = str_replace( 'http://', '', $this->transcoding_api_url );
+				$path_arr    = explode( '/', $path );
+				$server_addr = gethostbyname( $path_arr[0] );
+				set_transient( 'rtt_server_addr', $server_addr, 7 * 24 * 60 * 60 );
```
Comment (chandrapatel) [inline admin/rt-transcoder-handler.php]: Use time constants to set expiry time.
See: https://codex.wordpress.org/Transients_API#Using_Time_Constants

File: admin/rt-transcoder-handler.php [hunk 6]
```diff
@@ -991,6 +991,31 @@ public function handle_callback() {
...
+			if ( ! empty( $ignore_cache ) || false === $server_addr ) {
+
+				$path        = str_replace( 'http://', '', $this->transcoding_api_url );
+				$path_arr    = explode( '/', $path );
+				$server_addr = gethostbyname( $path_arr[0] );
+				set_transient( 'rtt_server_addr', $server_addr, WEEK_IN_SECONDS );
+
+			}
+
+			$incomming_addr = filter_input( INPUT_SERVER, 'REMOTE_ADDR' );
```
Comment (chandrapatel) [inline admin/rt-transcoder-handler.php]: Pass `FILTER_VALIDATE_IP` filter to validate. Also, please test it.

Comment (DevikVekariya) [reply admin/rt-transcoder-handler.php]: added `FILTER_VALIDATE_IP` to validate IP address. also tested it. it is working.

**Comment by juhi123** [comment]
Looks good 👍

**Pull request merged**